File: frontend/src/types/transcript.ts
export interface TranscriptSegment {
  id: string;
  assetId: string;
  speakerLabel: string | null;
  startTime: number;
  endTime: number;
  text: string;
  confidence: number | null;
  language: string | null;
}

export interface TranscriptResponse {
  segments: TranscriptSegment[];
  totalDuration: number;
  language: string | null;
  speakerCount: number;
}

export interface SceneInfo {
  id: string;
  assetId: string;
  sceneNumber: number;
  startTime: number;
  endTime: number;
  thumbnailUrl: string | null;
  duration: number;
}

export interface OcrRegion {
  id: string;
  assetId: string;
  frameNumber: number | null;
  timestamp: number | null;
  text: string;
  confidence: number | null;
}

export interface SearchResult {
  type: string;
  id: string;
  text: string;
  assetId: string | null;
  relevanceScore: number;
  metadata: Record<string, unknown>;
}

export interface SearchResponse {
  results: SearchResult[];
  total: number;
  facets: Record<string, number>;
}

export interface DuplicateCluster {
  id: string;
  caseId: string;
  status: string;
  members: DuplicateClusterMember[];
  createdAt: string;
}

export interface DuplicateClusterMember {
  id: string;
  assetId: string;
  originalFilename: string;
  distance: number | null;
  isPrimary: boolean;
}

File: backend/src/loom/schemas/transcript.py
from typing import Any
from uuid import UUID

from pydantic import BaseModel


class TranscriptSegmentResponse(BaseModel):
    """single transcript segment."""

    id: UUID
    asset_id: UUID
    speaker_label: str | None
    start_time: float
    end_time: float
    text: str
    confidence: float | None
    language: str | None
    model_name: str | None = None
    model_version: str | None = None
    model_params: dict[str, Any] | None = None

    model_config = {"from_attributes": True}


class TranscriptResponse(BaseModel):
    """full transcript for an asset."""

    segments: list[TranscriptSegmentResponse]
    total_duration: float
    language: str | None
    speaker_count: int
